File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/data/repository/SignUpRepositoryImpl.java
package com.bits.bee.bpmc.data.repository;

import java.lang.System;

/**
 * Created by aldi on 30/08/22.
 */
@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000&\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\u0018\u00002\u00020\u0001B\u000f\b\u0007\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004J\u001c\u0010\u0005\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\b0\u00070\u00062\u0006\u0010\t\u001a\u00020\nH\u0016R\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000\u00a8\u0006\u000b"}, d2 = {"Lcom/bits/bee/bpmc/data/repository/SignUpRepositoryImpl;", "Lcom/bits/bee/bpmc/domain/repository/SignUpRepository;", "apiUtils", "Lcom/bits/bee/bpmc/data/data_source/remote/ApiUtils;", "(Lcom/bits/bee/bpmc/data/data_source/remote/ApiUtils;)V", "postSignUp", "Lkotlinx/coroutines/flow/Flow;", "Lcom/bits/bee/bpmc/utils/Resource;", "Lcom/bits/bee/bpmc/data/data_source/remote/response/SignUpResponse;", "signUp", "Lcom/bits/bee/bpmc/domain/model/SignUp;", "app_debug"})
public final class SignUpRepositoryImpl implements com.bits.bee.bpmc.domain.repository.SignUpRepository {
    private final com.bits.bee.bpmc.data.data_source.remote.ApiUtils apiUtils = null;
    
    @javax.inject.Inject()
    public SignUpRepositoryImpl(@org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.data.data_source.remote.ApiUtils apiUtils) {
        super();
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public kotlinx.coroutines.flow.Flow<com.bits.bee.bpmc.utils.Resource<com.bits.bee.bpmc.data.data_source.remote.response.SignUpResponse>> postSignUp(@org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.domain.model.SignUp signUp) {
        return null;
    }
}

File: app/src/main/java/com/bits/bee/bpmc/data/repository/SignUpRepositoryImpl.kt
package com.bits.bee.bpmc.data.repository

import com.bits.bee.bpmc.data.data_source.remote.ApiUtils
import com.bits.bee.bpmc.data.data_source.remote.post.SendOtpPost
import com.bits.bee.bpmc.data.data_source.remote.post.SetupPasswordPost
import com.bits.bee.bpmc.data.data_source.remote.post.VerifDbPost
import com.bits.bee.bpmc.data.data_source.remote.post.VerifSmsPost
import com.bits.bee.bpmc.data.data_source.remote.response.*
import com.bits.bee.bpmc.domain.mapper.SignUpDataMapper
import com.bits.bee.bpmc.domain.model.SignUp
import com.bits.bee.bpmc.domain.repository.SignUpRepository
import com.bits.bee.bpmc.utils.ApiResponse
import com.bits.bee.bpmc.utils.NetworkBoundResource
import com.bits.bee.bpmc.utils.Resource
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

/**
 * Created by aldi on 30/08/22.
 */
class SignUpRepositoryImpl @Inject constructor(
    private val apiUtils: ApiUtils
) : SignUpRepository {

    override fun postSignUp(signUp: SignUp): Flow<Resource<SignUpResponse>> {
        return object : NetworkBoundResource<SignUpResponse>(){
            override fun createCall(): Flow<ApiResponse<SignUpResponse>> {
                return apiUtils.getSignUpApiService().postRegister(SignUpDataMapper.fromDomainToNetwork(signUp))
            }
        }.getAsFlow()
    }

    override fun postVerifSms(code: String, regId: Int): Flow<Resource<VerifSmsResponse>> {
        val verifSmsPost = VerifSmsPost(code = code, regid = regId, status = true)
        return object : NetworkBoundResource<VerifSmsResponse>(){
            override fun createCall(): Flow<ApiResponse<VerifSmsResponse>> {
                return apiUtils.getSignUpApiService().postVerifSms(verifSmsPost)
            }
        }.getAsFlow()
    }

    override fun postSendOtp(otpTipe: String, regId: Int): Flow<Resource<SendOtpResponse>> {
        val verifSmsPost = SendOtpPost(otpTipe, regId.toString())
        return object : NetworkBoundResource<SendOtpResponse>(){
            override fun createCall(): Flow<ApiResponse<SendOtpResponse>> {
                return apiUtils.getSignUpApiService().postSendOtp(verifSmsPost)
            }
        }.getAsFlow()
    }

    override fun postVerifDb(serial: String): Flow<Resource<CheckDbResponse>> {
        val verifSmsPost = VerifDbPost(serial = serial)
        return object : NetworkBoundResource<CheckDbResponse>(){
            override fun createCall(): Flow<ApiResponse<CheckDbResponse>> {
                return apiUtils.getSignUpApiService().postVerifDb(verifSmsPost)
            }
        }.getAsFlow()
    }

    override fun postSetupPassword(
        authKey : String,
        password: String,
        pin: String
    ): Flow<Resource<SetupPasswordResponse>> {
        val post = SetupPasswordPost(true, authKey, password, pin)
        return object : NetworkBoundResource<SetupPasswordResponse>(){
            override fun createCall(): Flow<ApiResponse<SetupPasswordResponse>> {
                return apiUtils.getSignUpApiService().postSetupPassword(post)
            }
        }.getAsFlow()
    }

    override fun getBidangUsaha(): Flow<Resource<BidangUsahaResponse>> {
        return object : NetworkBoundResource<BidangUsahaResponse>() {
            override fun createCall(): Flow<ApiResponse<BidangUsahaResponse>> {
                return apiUtils.getSignUpApiService().getBidangUsaha()
            }
        }.getAsFlow()
    }

}